internal/clients/ec2/aws_supported_types.go

```diff
@@ -0,0 +1,45 @@
+package ec2
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/payloads"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+func FilterAWSInstanceTypes(_ context.Context, it *[]clients.InstanceType) {
+	for _, i := range *it {
+		if IsSupported(i) {
+			// If the Instance type is present in the yml it is supported
+			i.Supported = true
+		}
+	}
+}
+
+func IsSupported(i clients.InstanceType) bool {
+	_, ok := clients.AwsSupported[i.Name]
+	return ok
+}
+
+func NewInstanceTypes(ctx context.Context, types []types.InstanceTypeInfo) ([]clients.InstanceType, error) {
+	list := make([]clients.InstanceType, 0, len(types))
+	for i := range types {
+		architectures := types[i].ProcessorInfo.SupportedArchitectures
+		for _, arch := range architectures {
+			arch, err := clients.MapArchitectures(ctx, string(arch))
+			if err != nil {
+				return nil, payloads.ClientError(ctx, "Instance type", "", err, 500)
+			}
+			list = append(list, clients.InstanceType{
+				Name:         string(types[i].InstanceType),
+				VCPUs:        types[i].VCpuInfo.DefaultVCpus,
+				Cores:        types[i].VCpuInfo.DefaultCores,
+				Memory:       *types[i].MemoryInfo.SizeInMiB,
+				Architecture: string(arch),
+				Supported:    false,
```

Comment:
We don't need the Filter method, if we do:
```suggestion
				Supported:    IsSupported(types[i].name),
```